ci-operator/step-registry/ibmcloud/provision/dedicatedhost/static/ibmcloud-provision-dedicatedhost-static-commands.sh

```diff
@@ -0,0 +1,143 @@
+## input parameter: the resource group of the dhg, dh profile, zone, name of the dh and the file which save the created dh name.
+function createDH() {
+    local rg=$1 dProfile=$2 zone=$3 dhName=$4 file=$5
+    local tmp family class dhg status
+    tmp=$(${IBMCLOUD_CLI} is dedicated-host-profile $dProfile --output JSON | jq -r '.family +","+ .class')
+    family=${tmp%,*}
+    class=${tmp#*,}
+    echo "family:" $family " class:" $class
+    dhg="${dhName}g"
+    run_command "${IBMCLOUD_CLI} is dedicated-host-group-create --zone $zone --family $family --class $class --name ${dhg}"
+    
+    ${IBMCLOUD_CLI} is dedicated-host-create --profile $dProfile --name $dhName --dhg $dhg --output JSON
+    
+    status=$(waitingStatus ${dhName})
+    echo "${dhName} status: ${status}"
+    run_command "${IBMCLOUD_CLI} is dedicated-host ${dhName}"
+
+    if [[ "${status}" != "stable" ]]; then
+        echo "ERROR: fail to create the dedicated host with profile $dProfile on zone: $zone resouce group:$rg"
+        echo "delete the dh ${dhName}, which status is ${status} ..."
+        run_command "${IBMCLOUD_CLI} is dhd ${dhName} -f"
+        run_command "${IBMCLOUD_CLI} is dhgd ${dhg} -f"
+    else
+        echo "${dhName}" > "${file}"        
+    fi
+}
+
+# waiting the pending dh status changed
+function waitingStatus() {
+    local dh=$1 status
+    COUNTER=0
+    while [ $COUNTER -lt 200 ]
+    do 
+        sleep 10
+        COUNTER=`expr $COUNTER + 20`
+        status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+        if [[ "${status}" != "pending" ]]; then
+            break
+        fi
+    done
+    echo ${status}
+}
+
+#####################################
+##############Initialize#############
+#####################################
+
+dhProfile="bx2-host-152x608"
+masterProfile="bx2-8x32"
+workerProfile="bx2-4x16"
+
+echo "the dh profile of the master nodes: ${dhProfile}; the dh profile of the worker nodes ${dhProfile}"
+
+echo "$(date -u --rfc-3339=seconds) - Creating the Dedicated Host..."
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+dhName="${CLUSTER_NAME}-dh"
+
+dhgRGFile=${SHARED_DIR}/ibmcloud_resource_group_dhg
+dh_file=${SHARED_DIR}/dedicated_host
+
+#####################################
+#######create the dedicated host#####
+#####################################
+ibmcloud_login
+
+#create the resouce group of the dedicated host group and saved in "${SHARED_DIR}/ibmcloud_resource_group_dhg"
+rg_dhg="${CLUSTER_NAME}-dhrg"
+
+echo "create resource group ... ${rg_dhg}"
+"${IBMCLOUD_CLI}" resource group-create ${rg_dhg} || exit 1
+echo "${rg_dhg}" >  ${dhgRGFile}
+
+run_command "${IBMCLOUD_CLI} target -g $rg_dhg"
+
+zone="${region}-${DEDICATEDHOST_ZONE}"
+echo "use the zone defined with DEDICATEDHOST_ZONE: ${zone}"
+
+createDH ${rg_dhg} ${dhProfile} ${zone} ${dhName} ${dh_file}
+
+
+if [ ! -f ${dh_file} ]; then
```

Comment:
I think we should not use `dh_file` as a judgement to decide if `createDH` pass or not, but `dh_file` should be used to save your dh info (whatever creation failed or passed), `createDH` should return `0` or non-zero to indicate it failed or not.